test/object-store/nested_collections.cpp

```diff
@@ -0,0 +1,186 @@
+    auto storage_list = dynamic_cast<Lst<Int>*>(collection_list1.get());
+    storage_list->add(5);
+    REQUIRE(storage_list->size() == 1);
+
+    // list of set
+    auto nested_obj2 = table2->create_object();
+    auto set_col_key = table2->get_column_key("nested_set");
+    auto list2 = nested_obj2.get_collection_list(set_col_key);
+    CHECK(list2->is_empty());
+    auto collection_set = list2->insert_collection(0);
+    auto storage_set = dynamic_cast<Set<Int>*>(collection_set.get());
+    storage_set->insert(5);
+    REQUIRE(storage_set->size() == 1);
+
+    // list of list of list
+    auto nested_obj3 = table3->create_object();
+    auto list_list_col_key = table3->get_column_key("nested_list_list");
+    auto list3 = nested_obj3.get_collection_list(list_list_col_key);
+    CHECK(list3->is_empty());
+    auto collection_list3 = list3->insert_collection_list(0);
+    auto collection3 = collection_list3->insert_collection(0);
+    auto storage_list3 = dynamic_cast<Lst<Int>*>(collection3.get());
+    storage_list3->add(5);
+    REQUIRE(storage_list3->size() == 1);
+    REQUIRE(collection_list3->size() == 1);
+
+    // list of dictionary of list
+    auto nested_obj4 = table4->create_object();
+    auto nested_dict_col_key = table4->get_column_key("nested_dict_list");
+    REQUIRE(table4->get_nesting_levels(nested_dict_col_key) == 2);
+    auto list4 = nested_obj4.get_collection_list(nested_dict_col_key);
+    CHECK(list4->is_empty());
+    auto collection4_dict = list4->insert_collection_list(0);
+    auto collection4 = collection4_dict->insert_collection("Test");
+    auto storage_list4 = dynamic_cast<Lst<Int>*>(collection4.get());
+    storage_list4->add(5);
+    REQUIRE(storage_list4->size() == 1);
+    REQUIRE(collection4->size() == 1);
+    REQUIRE(collection4_dict->size() == 1);
+
+    r->commit_transaction();
+}
+
+TEST_CASE("nested-dictionary", "[nested-collections]") {
+    InMemoryTestFile config;
+    config.cache = false;
+    config.automatic_change_notifications = false;
+    auto r = Realm::get_shared_realm(config);
+    r->update_schema({
+        {"dictionary_of_list",
+         {{"nested_list", PropertyType::Array | PropertyType::Int, {CollectionType::Dictionary}}}},
+        {"dictionary_of_set", {{"nested_set", PropertyType::Set | PropertyType::Int, {CollectionType::Dictionary}}}},
+        {"dictionary_of_list_of_dictionary",
+         {{"nested_list_dict",
+           PropertyType::Dictionary | PropertyType::Int,
+           {
+               CollectionType::Dictionary,
+               CollectionType::List,
+           }}}},
+    });
+    auto table1 = r->read_group().get_table("class_dictionary_of_list");
+    auto table2 = r->read_group().get_table("class_dictionary_of_set");
+    auto table3 = r->read_group().get_table("class_dictionary_of_list_of_dictionary");
+    REQUIRE(table1);
+    REQUIRE(table2);
+    REQUIRE(table3);
+
+    // TODO use object store API
```

Comment:
TODO

Reply:
This PR is only for schema support. The ObjectStore CRUD interface is not yet defined.